File: apps/web/test/event-labels.test.mjs
import assert from 'node:assert/strict';
import test from 'node:test';
import { eventPhaseLabel, eventRegistrationLabel } from '../lib/event-labels.ts';

test('event phases are readable and unknown dates are not treated as upcoming', () => {
  assert.equal(eventPhaseLabel('UPCOMING'), 'Upcoming');
  assert.equal(eventPhaseLabel('COMPLETED'), 'Past event');
  assert.equal(eventPhaseLabel('UNSCHEDULED'), 'Date to be announced');
  assert.equal(eventPhaseLabel(null), 'Date not confirmed');
});

test('stale registration status does not advertise past or undated entry', () => {
  assert.equal(eventRegistrationLabel('COMPLETED', 'OPEN'), undefined);
  assert.equal(eventRegistrationLabel('UNSCHEDULED', 'OPEN'), undefined);
  assert.equal(eventRegistrationLabel('UPCOMING', 'OPEN'), 'Registration listed open');
});

File: apps/fitcalgary_app/test/event_labels_test.dart
import 'package:fitcalgary_app/features/events/event_labels.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('event phases communicate past and unconfirmed dates', () {
    expect(eventPhaseLabel('UPCOMING'), 'UPCOMING');
    expect(eventPhaseLabel('COMPLETED'), 'PAST EVENT');
    expect(eventPhaseLabel('UNSCHEDULED'), 'DATE TO BE ANNOUNCED');
    expect(eventPhaseLabel(null), 'DATE NOT CONFIRMED');
  });

  test('stale registration state is not shown for old or undated events', () {
    expect(eventRegistrationLabel('COMPLETED', 'OPEN'), isNull);
    expect(eventRegistrationLabel('UNSCHEDULED', 'OPEN'), isNull);
    expect(
      eventRegistrationLabel('UPCOMING', 'OPEN'),
      'REGISTRATION LISTED OPEN',
    );
  });
}
